### PR TITLE
Make outer maximum startup timeout in `DockerComposeContainer` configurable

### PR DESCRIPTION
**Pull request by henri-tremblay**

### CODE REVIEW

File: core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java [hunk 1]
```diff
@@ -103,6 +103,8 @@
 
     private final Map<String, WaitAllStrategy> waitStrategyMap = new ConcurrentHashMap<>();
 
+    private Duration maximumTimeout = Duration.ofMinutes(30);
```
Comment (eddumelendez) [inline core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java:106]: let's rename it to `startupTimeout`

File: core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java [hunk 2]
```diff
@@ -598,6 +600,16 @@ public SELF withRemoveImages(RemoveImages removeImages) {
         return self();
     }
 
+    /**
+     * Set the maximum timeout all the waits set are bounded to.
+     *
+     * @return this instance. for chaining
+     */
+    public SELF withMaximumTimeout(Duration maximumTimeout) {
```
Comment (eddumelendez) [inline core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java:608]: let's rename it to `withStartupTimeout`

**Review by eddumelendez — CHANGES_REQUESTED** [review]
Thanks for the pr @henri-tremblay ! I have left some comments

File: core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java (hunk 2 above)
Comment (henri-tremblay) [reply core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java]: Done

File: core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java (hunk 1 above)
Comment (henri-tremblay) [reply core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java]: Done

**Review by eddumelendez — APPROVED** [review]

**Comment by kiview** [comment]
Thanks for your contribution @henri-tremblay 🙂

**Comment by henri-tremblay** [comment]
You are welcome. Thanks for merging it.

**Pull request merged**